numpy/core/src/multiarray/methods.c

```diff
@@ -858,20 +858,30 @@ array_astype(PyArrayObject *self,
      * and it's not a subtype if subok is False, then we
      * can skip the copy.
      */
-    if (!forcecopy && (order == NPY_KEEPORDER ||
-                       (order == NPY_ANYORDER &&
-                            (PyArray_IS_C_CONTIGUOUS(self) ||
-                            PyArray_IS_F_CONTIGUOUS(self))) ||
-                       (order == NPY_CORDER &&
-                            PyArray_IS_C_CONTIGUOUS(self)) ||
-                       (order == NPY_FORTRANORDER &&
-                            PyArray_IS_F_CONTIGUOUS(self))) &&
-                    (subok || PyArray_CheckExact(self)) &&
-                    PyArray_EquivTypes(dtype, PyArray_DESCR(self))) {
+    if (forcecopy != NPY_COPY_ALWAYS && 
+        (order == NPY_KEEPORDER ||
+        (order == NPY_ANYORDER &&
+            (PyArray_IS_C_CONTIGUOUS(self) ||
+            PyArray_IS_F_CONTIGUOUS(self))) ||
+        (order == NPY_CORDER &&
+            PyArray_IS_C_CONTIGUOUS(self)) ||
+        (order == NPY_FORTRANORDER &&
+            PyArray_IS_F_CONTIGUOUS(self))) &&
+    (subok || PyArray_CheckExact(self)) &&
+    PyArray_EquivTypes(dtype, PyArray_DESCR(self))) {
```

Comment:
NIT: can't say I am happy about that formatting (I should be more relaxed about style, but we end up at a higher level here)?
Most places in NumPy indent the following lines in the `if` by 8 spaces (and than additional if you open a bracket).